sunpy/coordinates/tests/test_wcs_utils.py

```diff
@@ -265,3 +268,103 @@ def test_self_observer():
     assert u.allclose(wcs.wcs.aux.hglt_obs, frame.lon.to_value(u.deg))
     assert u.allclose(wcs.wcs.aux.crln_obs, frame.lon.to_value(u.deg))
     assert u.allclose(wcs.wcs.aux.dsun_obs, frame.radius.to_value(u.m))
+
+
+@pytest.fixture
+def dkist_location():
+    return EarthLocation(*(-5466045.25695494, -2404388.73741278, 2242133.88769004) * u.m)
+
+
+@pytest.mark.remote_data
+def test_obsgeo_frame_mapping_cartesian(dkist_location, caplog):
+
+    obstime = Time("2021-05-21T03:00:00")
+    wcs = WCS(naxis=2)
+    wcs.wcs.ctype = ['HPLT', 'HPLN']
+    wcs.wcs.obsgeo = list(dkist_location.to_value(u.m).tolist()) + [0, 0, 0]
```

Comment:
```suggestion
    wcs.wcs.obsgeo[:3] = list(dkist_location.to_value(u.m).tolist())
```
is clearer, and should keep the default zero values in the second half of the array.

Reply:
It doesn't work, wcslib expects a 6 element array to be parsed to it. It doesn't default to an empty array it defaults to an unset pointer. (null)

Reply:
😢

Reply:
lol quite.